src/axom/quest/detail/shaping/PrimitiveSampler.hpp

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * \file PrimitiveSampler.hpp
+ *
+ * \brief Helper class for sampling-based shaping queries using primal geometric primitives
+ */
+
+#ifndef AXOM_QUEST_PRIMITIVE_SAMPLER__HPP_
+#define AXOM_QUEST_PRIMITIVE_SAMPLER__HPP_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/mint.hpp"
+#include "axom/spin.hpp"
+#include "axom/quest/detail/shaping/shaping_helpers.hpp"
+
+#include "axom/fmt.hpp"
+
+#include "mfem.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace shaping
+{
+using QFunctionCollection = mfem::NamedFieldsMap<mfem::QuadratureFunction>;
+using DenseTensorCollection = mfem::NamedFieldsMap<mfem::DenseTensor>;
+
+template <int NDIMS, typename ExecSpace>
+class PrimitiveSampler
+{
+public:
+  static constexpr int DIM = NDIMS;
+  using BVHType = spin::BVH<DIM, ExecSpace, double>;
+  using GeometricBoundingBox = primal::BoundingBox<double, DIM>;
+  using SpacePt = primal::Point<double, DIM>;
+
+  using SimplexType =
+    std::conditional_t<NDIMS == 3, primal::Tetrahedron<double, 3>, primal::Triangle<double, 2>>;
+
+  static_assert(NDIMS == 3 || NDIMS == 2, "PrimitiveSampler needs to be 2 or 3");
+
+public:
+  /**
+    * \brief Constructor for a PrimitiveSampler over a collection of triangles (in 2D) or tetrahedra (in 3D)
+    *
+    * \param shapeName The name of the shape; will be used for the field for the associated samples
+    * \param surfaceMesh Pointer to the surface mesh
+    *
+    * \note Does not take ownership of the surface mesh
+    */
+  PrimitiveSampler(const std::string& shapeName, std::shared_ptr<mint::Mesh> surfaceMesh)
+    : m_shapeName(shapeName)
+    , m_surfaceMesh(surfaceMesh)
+  { }
+
+  ~PrimitiveSampler() { }
+
+  std::shared_ptr<mint::Mesh> getSurfaceMesh() const { return m_surfaceMesh; }
+
+  /// Computes the bounding box of the surface mesh
+  void computeBounds()
```

Comment:
Could this work to compute the bounds of the mint mesh be moved into the mint mesh itself and then just return a bbox? Seeing it here, I'm wondering why the mint mesh can't do this for itself.

Reply:
Presumably, it already does.

I was planning to update this part of the class a bit so we filter out tets that are not in the bounding box of the current domain, which should help reduce the memory requirements.

Reply:
I'm going to leave this one as is since I'm planning to update the bounding box computation in a follow-up PR.